src/Service/CacheService.php

```diff
@@ -107,7 +107,10 @@ public function save(MvcEvent $mvcEvent)
 
         $cacheStorage = $this->getCacheStorage();
         if ($cacheStorage instanceof TaggableInterface) {
-            $tags = array_unique(array_merge($this->getTags($mvcEvent), $cacheEvent->getTags()));
+            $tags = array_map(
```

Comment:
Last optimization: Can you extract this to a private method `prefixTags(array $tags)`, because the same code is also in the `clearByTags()` method, so we can keep the codebase DRY this way.